packages/playwright/src/worker/workerMain.ts

```diff
@@ -191,14 +197,19 @@ export class WorkerMain extends ProcessRunner {
       return;
 
     this._config = await deserializeConfig(this._params.config);
-    this._project = this._config.projects.find(p => p.id === this._params.projectId)!;
+    const project = this._config.projects.find(p => p.id === this._params.projectId);
+    if (!project) {
+      this._isMissingProject = true;
```

Comment:
If you assign `this._config` after this check, we can get rid of `_isMissingProject` and replace it with `if (!!this._config)`. This would mean "if we never loaded config, there is nothing to cleanup anyway".